src/spid-sdk.js

```diff
@@ -1,165 +1,171 @@
-        _defaults = {
-            server: null,
-            client_id: null,
 
-            logging: false,
-            prod: true,
-            https: true,
-            storage: 'localstorage',
-            timeout: 15000,
-            refresh_timeout: 900000,
-            varnish_expiration: null
-        },
-        _options = {},
-        _initiated = false,
-        _session = {};
-
-    function init(options, callback) {
-        _options = this.Util.copy(options, _defaults);
-        if(!_options['server']) { throw new TypeError('[SPiD] server parameter is required'); }
-        if(!_options['client_id']) { throw new TypeError('[SPiD] client_id parameter is required'); }
-
-        //Set minimum refresh timeout
-        if(_options.refresh_timeout <= 60000) {
-            _options.refresh_timeout = 60000;
-        }
+function globalExport(global) {
+    global.SPiD = global.SPiD || this;
+    global.SPiD.Talk = require('./spid-talk');
+}
 
-        _initiated = true;
-        if(callback) {
-            callback();
-        }
+function init(opts, callback) {
+    config.init(opts);
+    if(!config.options().noGlobalExport) {
+        globalExport(window);
     }
-
-    function hasSession(callback) {
-        callback = callback || function() {};
-        var that = this,
-            respond = function(err, data) {
-                if(that.EventTrigger) {
-                    that.EventTrigger.session(_session, data);
-                }
-                _session = data;
-                callback(err, data);
-            },
-            handleResponse = function(err, data) {
-                if(that.Persist && !err && !!data.result) {
-                    that.Persist.set("Session", data, data.expiresIn);
-                }
-                respond(err, data);
-            },
-            handleException = function(err, data) {
-                if(err && err.type === "LoginException") {
-                    //Fallback to core
-                    return that.Talk.request(that.coreEndpoint(), null, {autologin:1}, handleResponse);
-                }
-                handleResponse(err, data);
-            };
-
-        if(this.Persist) {
-            var data = this.Persist.get("Session");
-            if(data) {
-                _session = data;
-                return respond(null, data);
-            }
-        }
-        this.Talk.request(this.sessionEndpoint(), null, {autologin:1}, handleException);
+    _initiated = true;
+    if(callback) {
+        callback();
     }
+}
 
-    function hasProduct(productId, callback) {
-        var cache = this.Persist,
-            util = this.Util;
-        callback = callback || function() {};
-        if(cache) {
-            var cacheVal = cache.get('prd_{id}'.replace('{id}', productId));
-            if(cacheVal && (cacheVal.refreshed + _options.refresh_timeout) > util.now()) {
-                return callback(null, cacheVal);
+function hasSession(callback) {
```

Comment:
I was thinking: do we need/want some way to bypass the cache mechanism here, perhaps via an additional parameter, @joawan ?